client/app/tasks/hooks/useEditTask.ts

```diff
@@ -0,0 +1,61 @@
+import { useRouter } from 'next/navigation'
+import { useQueryClient } from '@tanstack/react-query'
+import { useForm } from 'react-hook-form'
+import { useToast } from '@/_hooks'
+import { TaskType } from '@/_types'
+import { isErrorWithMessage } from '@/_utils'
+import { useEditTaskMutation } from './api'
+import { taskResolver, taskSchema } from '../schema'
+
+export const useEditTask = (defaultData: TaskType) => {
+  const queryClient = useQueryClient()
+  const { toast } = useToast()
+  const router = useRouter()
+
+  const { mutate, isPending } = useEditTaskMutation()
+
+  const form = useForm<taskSchema>({
+    resolver: taskResolver,
+    defaultValues: {
+      title: defaultData.title,
+      date: defaultData.date,
+      assigneeId: defaultData.assigneeId,
+      note: defaultData.note,
+    },
+  })
+
+  const handleSuccess = () => {
+    toast({ variant: 'success', title: 'Successfully edited a task!' })
+    queryClient.invalidateQueries({
+      queryKey: ['todos', 'me', 'house', 'tasks'],
```

Comment:
Hmm. Do you also need to update me? I feel it's too much dependencies 🤔  Best practice is less dependencies... 
Btw, if you update one task, you want to just delete the case for one task Id!

Reply:
Fixed it!
https://github.com/HAK2024/fairy-share/pull/84/commits/0ad6a929330779aa97859f5327cc043d5ab0a2df